### PR TITLE
Add permissions to GitHub actions

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Pinned actions by SHA https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies
- Included permissions for the action. https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions

>Pin actions to a full length commit SHA

>Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload.

https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

Also, dependabot supports upgrading based on SHA.

Signed-off-by: naveensrinivasan <172697+naveensrinivasan@users.noreply.github.com>

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mraible** [comment]
Why do you think this is a good idea? 

I'd prefer to use versions like `v2` and `v3` so minor releases are picked up without further maintenance. Pinning to commits seems like it'd be even more work than using patch versions.

**Comment by mshima** [comment]
Using commits is more secure if we did third pard actions audits.
We don't audit GitHub actions and we have dependabot auto merge.
So IMO is not really useful.

Action permissions is welcome.

**Comment by naveensrinivasan** [comment]
> Using commits is more secure if we did third pard actions audits. We don't audit GitHub actions and we have dependabot auto merge. So IMO is not really useful.
> 
> Action permissions is welcome.

Thanks, I would appreciate it if this could be merged sooner as it would avoid my re-work merge conflicts.

**Comment by mshima** [comment]
@naveensrinivasan the idea is to keep using tags as sha will make the maintenance harder.

If you revert the actions to use tags, we can merge the permissions.

**Comment by naveensrinivasan** [comment]
> @naveensrinivasan the idea is to keep using tags as sha will make the maintenance harder.
> 
> If you revert the actions to use tags, we can merge the permissions.

OK, I understand, I can do that.

**Comment by naveensrinivasan** [comment]
> @naveensrinivasan the idea is to keep using tags as sha will make the maintenance harder.
> 
> If you revert the actions to use tags, we can merge the permissions.

@mshima  I have updated the PR with just permissions. PTAL.

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -4,8 +4,14 @@ on:
...
       - 'dependabot/**'
 
+permissions:
+  contents: read
+
 jobs:
   triage:
+    permissions:
+      contents: read  # for actions/labeler to determine modified files
+      pull-requests: write  # for actions/labeler to add labels to PRs
```
Comment (mshima) [inline .github/workflows/labeler.yml:14]:
```suggestion
      contents: read # for actions/labeler to determine modified files
      pull-requests: write # for actions/labeler to add labels to PRs
```

File: .github/workflows/lock-maintenance.yml [hunk 2]
```diff
@@ -4,8 +4,14 @@ on:
...
   workflow_dispatch:
 
+permissions:
+  contents: read
+
 jobs:
   build:
+    permissions:
+      contents: write  # for peter-evans/create-pull-request to create branch
+      pull-requests: write  # for peter-evans/create-pull-request to create a PR
```
Comment (mshima) [inline .github/workflows/lock-maintenance.yml:14]:
```suggestion
      contents: write # for peter-evans/create-pull-request to create branch
      pull-requests: write # for peter-evans/create-pull-request to create a PR
```

File: .github/workflows/release-drafter.yml [hunk 3]
```diff
@@ -25,8 +25,14 @@ on:
...
       - main
 
+permissions:
+  contents: read
+
 jobs:
   update_release_draft:
+    permissions:
+      contents: write  # for release-drafter/release-drafter to create a github release
+      pull-requests: write  # for release-drafter/release-drafter to add label to PR
```
Comment (mshima) [inline .github/workflows/release-drafter.yml:35]:
```suggestion
      contents: write # for release-drafter/release-drafter to create a github release
      pull-requests: write # for release-drafter/release-drafter to add label to PR
```

**Review by mshima — COMMENTED** [review]
Prettier

**Comment by mshima** [comment]
@naveensrinivasan should we add
```
permissions:
   contents: read
```

to angular, react and vue workflows?

**Comment by naveensrinivasan** [comment]
> @naveensrinivasan should we add
> 
> ```
> 
> permissions:
> 
>    contents: read
> 
> ```
> 
> 
> 
> to angular, react and vue workflows?

Can that be a separate PR? Need to analyze before answering the question.

**Review by mshima — APPROVED** [review]

**Comment by mshima** [comment]
Thanks

**Pull request merged**